modules/core/src/main/scala/ch/epfl/scala/debugadapter/internal/evaluator/RuntimeDefaultValidator.scala

```diff
@@ -196,29 +194,24 @@ class RuntimeDefaultValidator(val frame: JdiFrame, val sourceLookUp: SourceLookU
       .orElse(ArrayElemTree(on, args))
 
   def validateIndirectApply(
-      on: Validation[RuntimeTree],
+      on: RuntimeTree,
       name: String,
       args: Seq[RuntimeEvaluableTree]
   ): Validation[RuntimeEvaluableTree] =
     for {
-      intermediate <- validateName(name, on, methodFirst = true).orElse(validateClass(name, on))
+      intermediate <- validateMember(name, on, methodFirst = true)
+        .orElse(localVarTreeByName(name))
```

Comment:
Do we still need this?

```suggestion
        .orElse(localVarTreeByName(name))
```

Reply:
Yes, because the "indirect" apply can be either on a field/method or on a local var (e.g. `list(0)`)

Reply:
Ok, I see